security/advancedtls/advancedtls.go

```diff
@@ -216,14 +216,24 @@ type ClientOptions struct {
 	// It could be nil if such checks are not needed.
 	RevocationConfig *RevocationConfig
 	// MinVersion contains the minimum TLS version that is acceptable.
-	// By default, TLS 1.2 is currently used as the minimum when acting as a
-	// client, and TLS 1.0 when acting as a server. TLS 1.0 is the minimum
-	// supported by this package, both as a client and as a server.
+	//
+	// Deprecated: use MinTLSVersion instead.
 	MinVersion uint16
 	// MaxVersion contains the maximum TLS version that is acceptable.
-	// By default, the maximum version supported by this package is used,
-	// which is currently TLS 1.3.
+	//
+	// Deprecated: use MinTLSVersion instead.
 	MaxVersion uint16
+	// MinTLSVersion contains the minimum TLS version that is acceptable.
+	// By default, TLS 1.2 is currently used as the minimum when acting as a
+	// client, and TLS 1.0 when acting as a server. TLS 1.0 is the minimum
+	// supported by this package, both as a client and as a server.  This
+	// default may be changed over time affecting backwards compatibility.
```

Comment:
Are we reserving the right to override the user's setting?  Like if they set 1.0 and we decide to set the min to 1.2, then we'll automatically upgrade them?  Or no?

Reply:
I don't think so - if we decide to change the minimum supported version from 1.0 to 1.2, I think we'd add in a check that throws an error rather than override the user setting.

Reply:
That's fine.  It seems like when they made a change like this in Go's tls package they added an env var so users could get the old 1.0-default behavior back (just FYI).

Also, I assume this is supposed to be a `tls.VersionTLSxx` value?  (I wish they had made a type name for it.)  That might be worth documenting.

Reply:
I think we could explore adding an env var like that down the line if it ever happens that we need it
Added notes in the docs about using tls.VersionTLSxx

Reply:
> Added notes in the docs about using tls.VersionTLSxx


What doc?  The godocs?  If so, I think you forgot to push the change.  It seems worth documenting in these comments, since this isn't inside the `tls` package, but I won't argue too strongly.

Reply:
Whoops I didn't actually push the change to my remote branch, done now